### PR TITLE
EVG-15413 Hide jira comment subscription option on waterfall page

### PR DESCRIPTION
**Pull request by bynn**
[EVG-15413](https://jira.mongodb.org/browse/EVG-15413)

### Description 
Disallow jira comment subscriptions to all tasks in a project

### Testing 
unit test

### CODE REVIEW

**Comment by ablack12** [comment]
Okay wow I didn't even know you could do this for the waterfall 😮 this looks like you're omitting it completely though; theoretically we only want to omit it for "any task finishes" and "any task fails" since those are the noisy ones (although if no one is using it we may as well omit it). 

We do also need to do this for the project page (this is actually where the issue came from originally).

File: model/event/subscriptions.go [hunk 1]
```diff
@@ -323,6 +323,11 @@ func (s *Subscription) Validate() error {
 	if !IsValidOwnerType(string(s.OwnerType)) {
 		catcher.Add(errors.Errorf("%s is not a valid owner type", s.OwnerType))
 	}
+	if s.OwnerType == OwnerTypeProject &&
+		s.ResourceType == ResourceTypeTask &&
+		s.Subscriber.Type == JIRACommentSubscriberType {
+		catcher.Add(errors.New("JIRA comment subscription not allowed for all tasks in the project"))
```
Comment (ablack12) [inline model/event/subscriptions.go:329]: We should probably disallow this for jira ticket creation as well. 

Also nit: catcher.New()

File: model/event/subscriptions.go [hunk 2]
```diff
@@ -323,6 +323,11 @@ func (s *Subscription) Validate() error {
 	if !IsValidOwnerType(string(s.OwnerType)) {
 		catcher.Add(errors.Errorf("%s is not a valid owner type", s.OwnerType))
 	}
+	if s.OwnerType == OwnerTypeProject &&
```
Comment (ablack12) [inline model/event/subscriptions.go:326]: Can you add a comment about why we're validating this?

**Review by ablack12 — CHANGES_REQUESTED** [review]

**Comment by ablack12** [comment]
Sorry if I made it sound like the waterfall change you made was wrong; we _should_ still disallow doing this from the waterfall level, I just didn't know that was a thing at the time!

File: model/event/subscriptions.go (hunk 1 above)
Comment (bynn) [reply model/event/subscriptions.go]: done and leaving the option in as discussed

File: model/event/subscriptions.go (hunk 2 above)
Comment (bynn) [reply model/event/subscriptions.go:327]: done

**Review by ablack12 — APPROVED** [review]

**Pull request merged**